pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -414,6 +455,73 @@ class AnalyticsImpl implements Analytics {
   @override
   void close() => _gaClient.close();
 
+  @override
+  Future<List<Survey>> fetchAvailableSurveys() async {
+    final surveysToShow = <Survey>[];
+    if (!okToSend) return surveysToShow;
+
+    final logFileStats = _logHandler.logFileStats();
+
+    // Call for surveys that have already been dismissed from
+    // persisted survey ids on disk
+    final persistedSurveyMap = _surveyHandler.fetchPersistedSurveys();
+
+    for (final survey in await _surveyHandler.fetchSurveyList()) {
+      // Apply the survey's sample rate; if the generated value from
+      // the client id and survey's uniqueId are less, it will not get
+      // sent to the user
+      if (survey.samplingRate < sampleRate(_clientId, survey.uniqueId)) {
+        continue;
+      }
+
+      // If the survey has been permanently dismissed or has temporarily
+      // been snoozed, skip it
+      if (surveySnoozedOrDismissed(survey, persistedSurveyMap)) continue;
+
+      // Counter to check each survey condition, if all are met, then
+      // this integer will be equal to the number of conditions in
+      // [Survey.conditionList]
+      var conditionsMet = 0;
+      for (final condition in survey.conditionList) {
+        if (logFileStats == null) continue;
+
+        // Retrieve the value from the [LogFileStats] with
+        // the label provided in the condtion
+        final logFileStatsValue =
+            logFileStats.getValueByString(condition.field);
+
+        if (logFileStatsValue == null) continue;
+
+        switch (condition.operatorString) {
+          case '>=':
+            if (logFileStatsValue >= condition.value) conditionsMet++;
+            break;
```

Comment:
You no longer need `break` statements in switch case: https://dart.dev/language/branches#switch-statements